cmd/serverless-init/initcontainer/initcontainer_test.go

```diff
@@ -63,3 +68,35 @@ func TestFlushTimeout(t *testing.T) {
 	assert.Equal(t, false, metricAgent.hasBeenCalled)
 	assert.Equal(t, false, mockLogsAgent.DidFlush())
 }
+
+func TestPropagateChildSuccess(t *testing.T) {
+	err := execute(&serverlessLog.Config{}, []string{"bash", "-c", "exit 0"})
+	assert.Equal(t, nil, err)
+}
+
+func TestPropagateChildError(t *testing.T) {
+	expectedError := 123
+	err := execute(&serverlessLog.Config{}, []string{"bash", "-c", "exit " + strconv.Itoa(expectedError)})
+	assert.Equal(t, expectedError<<8, int(err.(*exec.ExitError).ProcessState.Sys().(syscall.WaitStatus)))
+}
+
+func TestForwardSignalToChild(t *testing.T) {
+	resultChan := make(chan error)
+	terminatingSignal := syscall.SIGUSR1
+	cmd := exec.Command("sleep", "2s")
+	cmd.Start()
+	sigs := make(chan os.Signal, 1)
+	go forwardSignals(cmd.Process, &serverlessLog.Config{}, sigs)
```

Comment:
What happens if this command terminates before we call `forwardSignals` on it? Will the test fail? It's unlikely, but it could be a flaky test.

Reply:
I'd probably guess so, yeah.